### DISCUSSION TITLE
Versions (as of 2021-08-05) after initial release on 2021-07-12 are slower?

### DISCUSSION THREAD

**Discussion by colbyboles** [Chia-Network/bladebit#51] [General]
I have two identical EPYC 7713 machines with 512GB of DDR4 3200. One was running a build from your initial 2012-07-12 release. For the other I pulled the latest as of 2021-08-05 and built that. The machine with the newer code consistently takes around 7.08min per plot while the one with the initial code takes about 6.80min per plot. I built the newer code on the 6.80min machine and now it is also taking around 7.08min.  Both machines are using 120 of 128 threads of the CPU.

I can't think of why the bugfixes and NUMA improvements would actually make things slower? I may revert to the orginal code as I'm not having any issues with it and it seems to be about 4% faster.

**Comment by harold-b** [comment]
I would **not** use any commit before a990e2344e2fa228605e117c7e5bab87811cd0ea. There was a bug that generated an incorrect memo/id which would render pool plots invalid. So please do not use anything before that commit.

As for the difference in timings: This is likely due to the fix with CPU affinity. I assume half of your threads are logical threads and not actually physical cores. Because bladebit mostly saturates the cpu cores, there's barely any benefit on running on HT threads, in some cases it can potentially degrade performance. Which means that in the version previous to the update with the cpu affinity fix, the scheduler was doing taking good care of handing off work to the right cores.

That could*** be the case, perhaps not. But nothing else has changed that would degrade performance. The fix issued on 240a9b547736ea8d32b1998ba468f70c03ff2f3a does not degrade performance either, so that shouldn't be a problem. You could opt to not use NUMA memory interleaving as well by specifying the `--no-numa` parameter to see if it helps you.

**Reply by colbyboles** [reply]
Thanks. I will try --no-numa to see if that accounts for the difference. I can see in the numa version via htop that bladebit {-t 120} uses threads 0-119 and threads 120-127 are left for the system and other processes. In the older version the 8 "free" threads would jump around and not have any affinity. It is surprising that that behavior still resulted in faster plot times.

**Reply by icsta** [reply]
For anyone looking at this w/ an Epyc system. I tried the --no-numa flag and it decreased plotting speed a significant amount on a single Epyc 7371 server.
